be/src/pipeline/exec/union_source_operator.h

```diff
@@ -67,6 +68,49 @@ class UnionSourceOperator final : public SourceOperator<UnionSourceOperatorBuild
     std::shared_ptr<DataQueue> _data_queue;
     bool _need_read_for_const_expr;
 };
+class UnionSourceOperatorX;
+class UnionSourceLocalState final : public PipelineXLocalState<UnionDependency> {
+public:
+    ENABLE_FACTORY_CREATOR(UnionSourceLocalState);
+    using Base = PipelineXLocalState<UnionDependency>;
+    using Parent = UnionSourceOperatorX;
+    UnionSourceLocalState(RuntimeState* state, OperatorXBase* parent) : Base(state, parent) {};
+
+    Status init(RuntimeState* state, LocalStateInfo& info) override;
```

Comment:
[[nodiscard]]
we should make sure all status return value is checked

Reply:
this is a virtual function